dataplane/kernel-forwarder/pkg/kernelforwarder/common.go

```diff
@@ -0,0 +1,197 @@
+	dstName    string
+	srcIP      string
+	dstIP      string
+	srcIPVXLAN net.IP
+	dstIPVXLAN net.IP
+	vni        int
+}
+
+func setupLinkInNs(containerNs netns.NsHandle, ifaceName, ifaceIP string, inject bool) error {
+	if inject {
+		/* 1. Get a link object for the interface */
+		ifaceLink, err := netlink.LinkByName(ifaceName)
+		if err != nil {
+			logrus.Errorf("failed to get link for %q - %v", ifaceName, err)
+			return err
+		}
+		/* 2. Inject the interface into the desired container namespace */
+		if err = netlink.LinkSetNsFd(ifaceLink, int(containerNs)); err != nil {
+			logrus.Errorf("failed to inject %q in namespace - %v", ifaceName, err)
+			return err
+		}
+	}
+	/* 3. Lock the OS thread so we don't accidentally switch namespaces */
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+	/* 4. Save the current network namespace */
+	currentNs, err := netns.Get()
+	defer func() {
+		if err = currentNs.Close(); err != nil {
+			logrus.Error("error when closing:", err)
+		}
+	}()
+	if err != nil {
+		logrus.Errorf("failed to get current namespace: %v", err)
+		return err
+	}
+	/* 5. Switch to the new namespace */
+	if err = netns.Set(containerNs); err != nil {
+		logrus.Errorf("failed to switch to container namespace: %v", err)
+		return err
+	}
+	defer func() {
+		if err = containerNs.Close(); err != nil {
+			logrus.Error("error when closing:", err)
+		}
+	}()
+	/* 6. Get a link for the interface name */
+	link, err := netlink.LinkByName(ifaceName)
+	if err != nil {
+		logrus.Errorf("failed to lookup %q, %v", ifaceName, err)
+		return err
+	}
+	if inject {
+		var addr *netlink.Addr
+		/* 7. Parse the IP address */
+		addr, err = netlink.ParseAddr(ifaceIP)
+		if err != nil {
+			logrus.Errorf("failed to parse IP %q: %v", ifaceIP, err)
+			return err
+		}
+		/* 8. Set IP address */
+		if err = netlink.AddrAdd(link, addr); err != nil {
+			logrus.Errorf("failed to set IP %q: %v", ifaceIP, err)
+			return err
+		}
+		/* 9. Bring the interface UP */
+		if err = netlink.LinkSetUp(link); err != nil {
+			logrus.Errorf("failed to bring %q up: %v", ifaceName, err)
+			return err
+		}
+	} else {
+		/* 9. Bring the interface DOWN */
+		if err = netlink.LinkSetDown(link); err != nil {
+			logrus.Errorf("failed to bring %q down: %v", ifaceName, err)
+			return err
+		}
+		/* 2. Inject the interface back into the host namespace */
+		if err = netlink.LinkSetNsFd(link, int(currentNs)); err != nil {
+			logrus.Errorf("failed to inject %q bach to host namespace - %v", ifaceName, err)
+			return err
+		}
+	}
+	/* 10. Switch back to the original namespace */
+	if err = netns.Set(currentNs); err != nil {
+		logrus.Errorf("failed to switch back to original namespace: %v", err)
+		return err
+	}
+	return nil
+}
+
+//nolint
+func newConnectionConfig(crossConnect *crossconnect.CrossConnect, connType uint8) (*connectionConfig, error) {
+	switch connType {
+	case cLOCAL:
+		srcNsPath, err := crossConnect.GetLocalSource().GetMechanism().NetNsFileName()
```

Comment:
Could we have separate functions to create config in different cases?

Reply:
I prefer the idea of having a single point for getting the configuration, so I'm guessing you suggest to have a different function call instead of the code for each switch case, right? 

Something like:
```bash
case cLOCAL:
   newLocalConnectionConfig(....)
....
```

Reply:
Yes something like it, it could be done as separate PR I suppose.